core/src/main/java/org/elasticsearch/index/query/IdsQueryBuilder.java

```diff
@@ -144,14 +149,14 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
 
     @Override
     protected IdsQueryBuilder doReadFrom(StreamInput in) throws IOException {
-        IdsQueryBuilder idsQueryBuilder = new IdsQueryBuilder(in.readStringArray());
+        IdsQueryBuilder idsQueryBuilder = new IdsQueryBuilder(in.readOptionalStringArray());
```

Comment:
I made this change because we have an integration test which specifically checks that null Ids works. I would be for removing it but the fact that we have a specific test for it makes me think we rely on it somewhere?